traitsui/tests/editors/test_list_editor.py

```diff
@@ -0,0 +1,94 @@
+import copy
+import unittest
+
+from traits.api import HasStrictTraits, Instance, Int, List, Str
+from traitsui.api import Item, ListEditor, View
+from traitsui.testing.tester import command, locator, query
+from traitsui.testing.tester.exceptions import LocationNotSupported
+from traitsui.testing.tester.ui_tester import UITester
+from traitsui.tests._tools import (
+    requires_toolkit,
+    ToolkitName,
+)
+
+
+# 'Person' class:
+class Person(HasStrictTraits):
+
+    # Trait definitions:
+    name = Str()
+    age = Int()
+
+    # Traits view definition:
+    traits_view = View(
+        'name', 'age',
+        width=0.18,
+        buttons=['OK', 'Cancel']
+    )
+
+
+# Sample data:
+people = [
+    Person(name='Dave', age=39),
+    Person(name='Mike', age=28),
+    Person(name='Joe', age=34),
+    Person(name='Tom', age=22),
+    Person(name='Dick', age=63),
+    Person(name='Harry', age=46),
+    Person(name='Sally', age=43),
+    Person(name='Fields', age=31)
+]
+
+
+# 'ListTraitTest' class:
+class ListTraitTest(HasStrictTraits):
+
+    # Trait definitions:
+    people = List(Instance(Person, ()))
+    num_columns = Int(1)
+
+    # Traits view definitions:
+    def default_traits_view(self):
+        view = View(
+            Item(
+                'people',
+                label='List',
+                id='list',
+                style='custom',
+                editor=ListEditor(style='custom', columns=self.num_columns),
+            ),
+            resizable=True
+        )
+        return view
+
+
+@requires_toolkit([ToolkitName.qt, ToolkitName.wx])
+class TestCustomListEditor(unittest.TestCase):
+
+    def test_locate_element_and_edit(self):
+        # varying the number of columns in the view tests the logic for
+        # getting the correct nested ui
+        for col in range(1, 5):
+            obj = ListTraitTest(people=copy.deepcopy(people), num_columns=col)
+            tester = UITester()
+            with tester.create_ui(obj) as ui:
+                # sanity check
+                self.assertEqual(obj.people[7].name, "Fields")
+                people_list = tester.find_by_name(ui, "people")
+                item = people_list.locate(locator.Index(7))
+                name_field = item.find_by_name("name")
+                for _ in range(6):
+                    name_field.perform(command.KeyClick("Backspace"))
+                name_field.perform(command.KeySequence("David"))
```

Comment:
Minor: I'd have wanted to keep these two were it not because of the ongoing issue with KeyClick... Perhaps for the purpose of this PR we can just keep `DisplayedText()` in the next line. That would be enough to verify we have got the right entry.

Reply:
once #1184 is merged, this test should pass.  Will wait to merge this PR until after that has been merged.

Reply:
> once #1184 is merged, this test should pass. Will wait to merge this PR until after that has been merged.

It is possible that it would still if the insertion points start incorrectly.  1184 only made the change in default for range editor.  I could see an issue where the backspaces occur before Fields and we get an output of "FieldsDavid".  I am checking this now in a draft PR on a junk branch that I will delete after the CI is finished.  If a fix is needed for the insertion point stuff I will add it to this PR

Reply:
We don't know how long it'd take for #1184 to be merged. I'd still prefer decoupling this PR from #1184 so we can review and merge it more efficiently. But if you'd like to wait, that's okay for me.